src/vanilla/utils/atomWithLazy.ts

```diff
@@ -0,0 +1,20 @@
+import { atom } from '../../vanilla.ts'
+import type { Atom, PrimitiveAtom, SetStateAction } from '../../vanilla.ts'
+
+export function atomWithLazy<Value>(makeInitial: () => Value) {
+  const wrappedAtom: Atom<PrimitiveAtom<Value>> & { init?: Atom<undefined> } =
+    atom(() => atom(makeInitial()))
+
+  const proxyAtom = atom(
+    (get) => get(get(wrappedAtom)),
+    (get, set, value: SetStateAction<Value>) => set(get(wrappedAtom), value),
+  )
+
+  wrappedAtom.init = atom(undefined)
+  // when writing to wrappedAtom through proxyAtom, the store
+  // thinks we are actually storing the value of `proxyAtom`.
+  wrappedAtom.unstable_is = (a) =>
+    a.unstable_is ? a.unstable_is(proxyAtom) : a === proxyAtom
```

Comment:
It would be super nice if we had a test that fails without this. Is it too hard?

Reply:
I would be nice. It would require a barebones recreation of what `jotai-scope` is doing to intercept atoms. I wanted to understand how `jotai-scope` works better anyway, so I am happy to write it up. It will probably take a couple of days though.

Reply:
Although this hack works, I would like to better understand why it is necessary for `atomWithLazy` to function correctly with `jotai-scope`, both the `.init = atom(undefined)` and `unstable_is`.